### PR TITLE
Core:  + coPBftSigThd; optionalise coStaticKeySigningKeyFile & coStaticKey

### PR DESCRIPTION
**Pull request by deepfire**
1. add `coPBftSigThd` to `Core`
2. make `coStaticKeySigningKeyFile` and `coStaticKeyDlgCertFile` optional

### CODE REVIEW

**Comment by ksaric** [comment]
Please rebase on top of https://github.com/input-output-hk/cardano-shell/pull/237 since that will be the cleaner solution.

**Comment by deepfire** [comment]
@ksaric, I can't just yet, for demo purposes.

**Comment by deepfire** [comment]
@ksaric, updated.

File: src/Cardano/Shell/Constants/CLI.hs [hunk 1]
```diff
@@ -116,18 +116,20 @@ configCoreCLIParser = PartialCore
...
-           ( long "signing-key"
-          <> metavar "FILEPATH"
-          <> help "Path to the signing key."
-           )
-    <*> lastStrOption
-           ( long "delegation-certificate"
-          <> metavar "FILEPATH"
-          <> help "Path to the delegation certificate."
-           )
+    <*> ((Just <$>) <$> lastStrOption
```
Comment (ksaric) [inline src/Cardano/Shell/Constants/CLI.hs]: To be perfectly honest, I don't see the point of doing this.
Think about it, we already have a `Maybe`.

File: src/Cardano/Shell/Constants/CLI.hs [hunk 2]
```diff
@@ -116,18 +116,20 @@ configCoreCLIParser = PartialCore
...
+                <> help "Path to the signing key."
+                ))
+    <*> ((Just <$>) <$> lastStrOption
+         ( long "delegation-certificate"
+                <> metavar "FILEPATH"
+                <> help "Path to the delegation certificate."
+                ))
     <*> lastOption configNetworkMagicCLIParser
     <*> lastOption configDBVersionCLIParser
+    <*> lastOption (option auto $ long "pbft-signature-threshold" <> metavar "DOUBLE"
```
Comment (ksaric) [inline src/Cardano/Shell/Constants/CLI.hs]: Should use existing functions.

File: src/Cardano/Shell/Constants/PartialTypes.hs [hunk 3]
```diff
@@ -49,17 +49,12 @@ data PartialCardanoConfiguration = PartialCardanoConfiguration
...
-    -- ^ Static key signing file.
-    , pcoStaticKeyDlgCertFile       :: !(Last FilePath)
-    -- ^ Static key delegation certificate.
+    , pcoStaticKeySigningKeyFile    :: !(Last (Maybe FilePath))
+    , pcoStaticKeyDlgCertFile       :: !(Last (Maybe FilePath))
     , pcoRequiresNetworkMagic       :: !(Last RequireNetworkMagic)
-    -- ^ Do we require the network byte indicator for mainnet, testnet or staging?
     , pcoDBSerializeVersion         :: !(Last Integer)
-    -- ^ Versioning for values in node's DB.
+    , pcoPBftSigThd                 :: !(Last (Maybe Double))
```
Comment (ksaric) [inline src/Cardano/Shell/Constants/PartialTypes.hs]: This just looks weird.

Comment (dcoutts) [reply src/Cardano/Shell/Constants/PartialTypes.hs]: Yeah it doesn't need to be this way. It's already optional. It just means when we finalise the config, it's ok if it has no default value, we convert it into a `Maybe` at that point. Feel free to follow-up to fix this.

Comment (ksaric) [reply src/Cardano/Shell/Constants/PartialTypes.hs]: Will do.

Comment (deepfire) [reply src/Cardano/Shell/Constants/PartialTypes.hs]: The idea was to have a `Maybe` in the final `CardanoConfiguration`.  Now I'm struggling to remember why I thought it was needed.

Comment (deepfire) [reply src/Cardano/Shell/Constants/PartialTypes.hs]: Undoing in https://github.com/input-output-hk/cardano-shell/pull/246.

File: src/Cardano/Shell/Constants/CLI.hs (hunk 1 above)
Comment (deepfire) [reply src/Cardano/Shell/Constants/CLI.hs]: @ksaric, consider that we need to detect if the end user supplied just one of the certificate and the key -- that's an error.

If we don't have the `Maybe`, we wouldn't be able to detect it.

File: src/Cardano/Shell/Constants/PartialTypes.hs (hunk 3 above)
Comment (deepfire) [reply src/Cardano/Shell/Constants/PartialTypes.hs]: Yes, I remember now!

https://github.com/input-output-hk/cardano-shell/pull/246#issuecomment-518907922 is the explanation for `Maybe Double` in `coPbftSigThd`!

**Pull request merged**